docs: Security-related docs updates and lagging tasks (#26835)
MystenLabs/sui: docs/content/onchain-finance/asset-custody/address-balances/migrate-address-balances.mdx

1: 0x2::transfer::public_transfer(result(0))
```

### Compatibility coin reservations for legacy clients

> A JSON-RPC compatibility layer presents compatibility coin reservations to clients that cannot upgrade. A compatibility coin reservation is a synthetic object reference that reserves address balance funds for backward compatibility with clients that still expect coin object references. It is not a persisted `Coin<T>` object.

Reviewer: Also confirm with @mystenmark

> These reservations exist only to preserve basic functionality for older clients that expect coin object references. Do not treat them as normal persisted coin objects, and do not use them as a model for new integrations. For new development, use address balances directly, as described in [Using Address Balances](/onchain-finance/asset-custody/address-balances/using-address-balances).

Reviewer 2: ```suggestion
These reservations primarily to preserve basic functionality for older clients that expect coin object references. Do not treat them as normal persisted coin objects. For new development, use address balances directly, as described in [Using Address Balances](/onchain-finance/asset-custody/address-balances/using-address-balances).
```

> When a legacy client supplies a compatibility coin reservation as gas payment, the network resolves it as a withdrawal from the address balance rather than smashing it into a coin. Gas smashing applies only to real `Coin<SUI>` objects supplied as gas payment. For more information, see [Gas Smashing](/develop/transaction-payment/gas-smashing).

Reviewer 2: ```suggestion
When a legacy client supplies a compatibility coin reservation as gas payment, the network resolves it as a withdrawal from the address balance rather than smashing it into a coin. And gas smashing applies depending on the position of the coin reservation in the list of gas payments. For more information, see [Gas Smashing](/develop/transaction-payment/gas-smashing).
```
